### ISSUE TITLE
Detachable keyboard computer fails to log

### ISSUE THREAD

**Issue by JonnyMako** [kernc/logkeys#219]
I have a 2-in-1 computer which fails to log keys. Following the documentation of this project, I found `-d event24` fixes the problem. I'm reporting this issue per documentation. `/proc/bus/input/devices` attached.
[devices.txt](https://github.com/kernc/logkeys/files/4841010/devices.txt)

**Comment by JonnyMako** [comment]
Also is there a way to pass the device # to `llk` script?

**Comment by JonnyMako** [comment]
And unfortunately, the event # changes with every reboot. It would be nice if there was some way to figure this out. Happy to provide debug info.

**Comment by JonnyMako** [comment]
After thinking about this, the solution here would be to have an option for `logkeys` to identify devices by-id which are persistent and don't change across boots.

So, perhaps something like
```
logkeys -s -d-by-id  some-device-name
```

Alternatively if `-d` can take a relative path as input that should also work. In which case it would look like:

```
logkeys -s -d by-id/some-device-name
```
where the default location is `/dev/input` and then it would look in `by-id` directory for the device by id.

I prefer the second option since it gives us the flexibility to navigate to other device locations relative to `/dev/input`

**Comment by kernc** [comment]
Some related effort expended in https://github.com/kernc/logkeys/issues/161.

> Alternatively if `-d` can take a relative path as input that should also work.

Seems only a slight adjustment needed there. PR welcome.
https://github.com/kernc/logkeys/blob/0427dec4ecfd4e8a7844c3e7deb45404f0416cd3/src/logkeys.cc#L675-L677

**Comment by JonnyMako** [comment]
> Some related effort expended in #161.
> 
> > Alternatively if `-d` can take a relative path as input that should also work.
> 
> Seems only a slight adjustment needed there. PR welcome.
> 
> https://github.com/kernc/logkeys/blob/0427dec4ecfd4e8a7844c3e7deb45404f0416cd3/src/logkeys.cc#L675-L677

Absolutely, I got this working locally just last night. I'm looking at another issue where logkeys quits logging. I'm not sure what is causing it. It just issues a message that it has stopped logging. Either way, I'll open a PR soon-ish.
